docs/en/install-upgrade/installing-stack-demo-secure.asciidoc

```diff
@@ -0,0 +1,894 @@
++
+["source","shell"]
+----
+10.128.0.84
+----
+.. When prompted, confirm that the settings are correct.
+.. When prompted, choose to generate additional certificates, and then repeat the previous steps to add hostname and IP settings for each node in your {es} cluster.
+.. Provide a password for the generated `http.p12` keystore file.
+.. The generated files will be included in a zip archive. At the prompt, provide a path and filename for where the archive should be created.
++
+For this example we use: `/etc/elasticsearch/certs/elasticsearch-ssl-http.zip`:
++
+["source","shell"]
+----
+What filename should be used for the output zip file? [/usr/share/elasticsearch/elasticsearch-ssl-http.zip] /etc/elasticsearch/certs/elasticsearch-ssl-http.zip
+----
+
+. Earlier, when you generated the certificate for the transport layer, the default filename was `elastic-certificates.p12`. Now, when generating a certificate for the HTTP layer, the default filename is `http.p12`. This matches the name of the existing HTTP layer certificate file from when the initial {es} cluster was first installed.
++
+Just to avoid any possible name collisions, rename the existing `http.p12` file to distinguish it from the newly-created keystore:
++
+["source","shell"]
+----
+mv http.p12 http-old.p12
+----
+
+. Unzip the generated `elasticsearch-ssl-http.zip` archive:
++
+["source","shell"]
+----
+unzip elasticsearch-ssl-http.zip
+----
+
+. When the archive is unpacked, the certificate files are located in separate directories for each {es} node and for the {kib} node.
++
+You can run a recursive `ls` command to view the file structure:
++
+["source","shell"]
+----
+ls -lR elasticsearch kibana
+----
++
+["source","shell"]
+----
+elasticsearch:
+total 0
+drwxr-xr-x. 2 root root 56 Dec 12 19:13 mynode-es1
+drwxr-xr-x. 2 root root 72 Dec 12 19:04 mynode-es2
+drwxr-xr-x. 2 root root 72 Dec 12 19:04 mynode-es3
+
+elasticsearch/mynode-es1:
+total 8
+-rw-r--r--. 1 root root 1365 Dec 12 19:04 README.txt
+-rw-r--r--. 1 root root  845 Dec 12 19:04 sample-elasticsearch.yml
+
+elasticsearch/mynode-es2:
+total 12
+-rw-r--r--. 1 root root 3652 Dec 12 19:04 http.p12
+-rw-r--r--. 1 root root 1365 Dec 12 19:04 README.txt
+-rw-r--r--. 1 root root  845 Dec 12 19:04 sample-elasticsearch.yml
+
+elasticsearch/mynode-es3:
+total 12
+-rw-r--r--. 1 root root 3652 Dec 12 19:04 http.p12
+-rw-r--r--. 1 root root 1365 Dec 12 19:04 README.txt
+-rw-r--r--. 1 root root  845 Dec 12 19:04 sample-elasticsearch.yml
+
+kibana:
+total 12
+-rw-r--r--. 1 root root 1200 Dec 12 19:04 elasticsearch-ca.pem
+-rw-r--r--. 1 root root 1306 Dec 12 19:04 README.txt
+-rw-r--r--. 1 root root 1052 Dec 12 19:04 sample-kibana.yml
+----
+
+. In the directory where you unzipped the archive, replace your existing keystore with the new keystore. The location of your certificate directory may be different than what is shown here, depending on the installation method you chose.
```

Comment:
In keeping with the idea of full paths, I think we can remove 'In the directory where you unzipped the archive' and just start with 'Replace your existing keystore.." It will be obvious on the next step when using full paths.